### PR TITLE
Java: Add booking sample with M365 booking system

### PR DESCRIPTION
**Pull request by milderhc**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/samples/booking-agent-m365/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.microsoft.semantic-kernel</groupId>
+        <artifactId>semantickernel-parent</artifactId>
+        <version>1.0.1-SNAPSHOT</version>
```
Comment (johnoliver) [inline java/samples/booking-agent-m365/pom.xml:9]: will be at 1.0.2-SNAPSHOT now

File: java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import com.microsoft.semantickernel.semanticfunctions.annotations.DefineKernelFunction;
+import com.microsoft.semantickernel.semanticfunctions.annotations.KernelFunctionParameter;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+public class BookingPlugin {
+    public static final int BOOKING_HOURS = 2;
+    private GraphServiceClient graphServiceClient;
```
Comment (johnoliver) [inline java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java:20]: think can make these all final

File: java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                    type = OffsetDateTime.class)
+            OffsetDateTime date,
+            @KernelFunctionParameter(
+                    name = "partySize",
+                    description = "The number of people in the party",
+                    type = int.class)
+            int partySize
+    ) {
+        BookingAppointment bookingAppointment = new BookingAppointment();
+        bookingAppointment.setOdataType("#microsoft.graph.bookingAppointment");
```
Comment (johnoliver) [inline java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java:57]: are these fluent? i.e can you do:
 ```
 BookingAppointment bookingAppointment = new BookingAppointment()
        .setOdataType("#microsoft.graph.bookingAppointment")
        .setCustomerTimeZone(userTimeZone);
        .setSmsNotificationsEnabled(false);
 ```

**Review by dsgrieve — APPROVED** [review]

File: java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java (hunk 3 above)
Comment (milderhc) [reply java/samples/booking-agent-m365/src/main/java/com/microsoft/semantickernel/BookingPlugin.java:57]: no, they're not

**Pull request merged**